app/controllers/comments_controller.rb

```diff
@@ -1,6 +1,8 @@
 # frozen_string_literal: true
 
 class CommentsController < ApplicationController
+  before_action :find_comment, only: %i[edit update]
+
   def new
     @giveaway = Giveaway.find(params[:giveaway_id])
```

Comment:
To DRY the code, I believe we can extract this in a separate method `set_giveaway` and call it in a before action.